modules/redis/redis.c

```diff
@@ -87,21 +88,12 @@ redis_dd_set_auth(LogDriver *d, const gchar *auth)
 
 void
 redis_dd_set_command_ref(LogDriver *d, const gchar *command,
-                         LogTemplate *key,
-                         LogTemplate *param1, LogTemplate *param2)
+                         GList *fields)
 {
   RedisDriver *self = (RedisDriver *)d;
 
   g_string_assign(self->command, command);
-
-  log_template_unref(self->key);
-  self->key = key;
-
-  log_template_unref(self->param1);
-  self->param1 = param1;
-
-  log_template_unref(self->param2);
-  self->param2 = param2;
+  self->fields = fields;
```

Comment:
You should free `self->fields` before setting it into a new value.

Reply:
Thanks fixed!